src/ui/ui.d.ts

```diff
@@ -0,0 +1,6 @@
+interface AppState {
+  origin: string;
+  route: string;
+  isSSR: boolean;
+  appData: any;
+}
```

Comment:
should we name this file `app-state.d.ts` maybe for more clarity?

Reply:
Sure, that makes sense.